subxt/src/utils/unchecked_extrinsic.rs

```diff
@@ -107,3 +107,29 @@ impl<Address, Call, Signature, Extra> IntoVisitor
         UncheckedExtrinsicDecodeAsTypeVisitor(PhantomData)
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+
+    #[test]
+    fn unchecked_extrinsic_encoding() {
+        let tx_bytes = vec![1, 2, 3];
+
+        let unchecked_extrinsic = UncheckedExtrinsic::<(), (), (), ()>::new(tx_bytes.clone());
+        let encoded_tx_bytes = unchecked_extrinsic.encode();
+
+        // The encoded representation must not alter the provided bytes.
+        assert_eq!(tx_bytes, encoded_tx_bytes);
+
+        // However, for decoding we expect to be able to read the extrinsic from the wire
+        // which would be length prefixed.
+        let wire_bytes = tx_bytes.encode();
+        let decoded_tx =
+            UncheckedExtrinsic::<(), (), (), ()>::decode(&mut &wire_bytes[..]).unwrap();
+        let encoded_tx_bytes = decoded_tx.encode();
+
+        assert_eq!(decoded_tx.bytes(), encoded_tx_bytes);
+        assert_eq!(tx_bytes, encoded_tx_bytes);
```

Comment:
Nit: 

I think this does what I'm hoping but it's a touch hard for me to follow (this stuff is in general because we are talking about encoding vec-like bytes into another vec :D)

The test could start with:

```rust
// A tx is basically some bytes with a compact length prefix; ie an encoded vec:
let tx_bytes =  vec![1,2,3].encode();
```

And then we have a "valid" TX format that can be used for all the tests, rather than needing a `let wire_bytes = tx_bytes.encode();` to make it valid for the second half :)